demo/src/app.jsx

```diff
@@ -557,6 +576,62 @@ function AppContent({ language, onLanguageClick }) {
             <SnackWarningButton />
             <SnackInfoButton />
 
+            <Button
+                variant="contained"
+                style={{
+                    float: 'left',
+                    margin: '5px',
+                }}
+                onClick={() => setOpenMultiChoiceDialog(true)}
+            >
+                Checkbox list
+            </Button>
+            <MultipleSelectionDialog
+                options={checkBoxListOption}
+                selectedOptions={[]}
+                open={openMultiChoiceDialog}
+                getOptionLabel={(o) => o.label}
+                getOptionId={(o) => o.id}
+                handleClose={() => setOpenMultiChoiceDialog(false)}
+                handleValidate={() => setOpenMultiChoiceDialog(false)}
+                titleId="Checkbox list"
+                divider={false}
+                secondaryAction={secondaryAction}
+                addSelectAllCheckbox={false}
+            />
+
+            <Button
+                variant="contained"
+                style={{
+                    float: 'left',
+                    margin: '5px',
+                }}
+                onClick={() => setOpenDraggableMultiChoiceDialog(true)}
+            >
+                Draggable checkbox list
+            </Button>
+            <MultipleSelectionDialog
+                options={checkBoxListOption}
+                selectedOptions={[]}
+                open={openDraggableMultiChoiceDialog}
+                getOptionLabel={(o) => o.label}
+                getOptionId={(o) => o.id}
+                handleClose={() => setOpenDraggableMultiChoiceDialog(false)}
+                handleValidate={() => setOpenDraggableMultiChoiceDialog(false)}
+                titleId="Draggable checkbox list"
+                divider
+                secondaryAction={secondaryAction}
+                isDndDragAndDropActive
+                onDragEnd={({ source, destination }) => {
+                    if (destination !== null && source.index !== destination.index) {
+                        const res = [...checkBoxListOption];
+                        const [item] = res.splice(source.index, 1);
+                        res.splice(destination ? destination.index : checkBoxListOption.length, 0, item);
+                        setCheckBoxListOption(res);
+                    }
+                }}
+            />
```

Comment:
many of those props don't exist in MultipleSelectionDialogProps (divider, secondaryACtion, isDndDragAndDropActive,...)
improve MultipleSelectionDialogProps to make this component usable in a TS environment

Reply:
it is fixed by making MultipleSelectionDialogProps extend CheckboxListProps